src/server.py

```diff
@@ -0,0 +1,73 @@
+from __future__ import unicode_literals
+import socket
+
+
+class HTTPErrors(Exception):
+    """HTTP Error exception class."""
+
+    def __init__(self, message):
+        """Intialize Error with message."""
+        self.message = message
+
+
+def parse_request(request):
+    """Parse request and if valid return URI."""
+    split_req = request.split('\r\n', 1)
+    method, uri, proto = split_req[0].split()
+    headers = split_req[1].split('\r\n\r\n')
```

Comment:
consider limiting this split to 1 as well.  You don't really know anything about what might be in the body of the request.